### DISCUSSION TITLE
Should we move along the proposals to trim aspect evaluation.

### DISCUSSION THREAD

**Discussion by aiuto** [bazelbuild/bazel#22026] [Design proposals]
Last fall,  I (@aiuto) proposed two ideas to change aspect evaluation.
- [Faster aspects: pruning on the way down](https://docs.google.com/document/d/1Md4BX2v-VMV-qNmc-wdeImTCdBK9BRF6YNktmOBVVVY/edit) outlined how we could add a pre-traversal phase to aspect evaluation so that particular rule types could ignore some attributes. For example, in a `*` aspect evaluation, we might ignore a known tool on a particular kind of rule.  This would be used to prevent descent into potentially very large subtrees.
- [Better SBOMS: annotated rule attributes](https://docs.google.com/document/d/1ReiTFz5N98D9bGPgL45ziQnGT8xJAunug9JJ89gZsm8/edit) proposed adding labels/tags/something to rule attribute definitions and making those available at aspect evaluation time. Combined with the above, this could be used to make choices about what paths to descend.

There has been continued interest in these proposals, so I would like to move them along or bury them.  Towards moving them along I want to restart the discussion, with the first goal of getting an explicit approval from the Bazel team that it is OK to even think about this. Given that, we can work together on more specific designs. And, finally, we can see if anyone will find the cycles to build these features.

cc: @brentleyjones

**Comment by mai93** [comment]
I'm picking up this proposal, please check the new [design document](https://docs.google.com/document/d/1LfglQ4naytDXmvT2Y8ptnoWOv8awhuaToNGmD2dRGdU/edit?usp=sharing)

**Comment by aiuto** [comment]
Awesome!!!!
